python/servo/testing_commands.py

```diff
@@ -592,17 +602,18 @@ def parse_speedometer_result(result):
                     }
                 }
             else:
-                raise "Unknown unit!"
+                raise Exception("Unknown unit!")
 
             for child in result["children"]:
                 parse_speedometer_result(child)
 
         for v in speedometer.values():
             parse_speedometer_result(v)
-        with open(bmf_output, "w", encoding="utf-8") as f:
-            json.dump(output, f, indent=4)
+        if bmf_output is not None:
```

Comment:
This check is not needed if we change the type of `bmf_output` to just a `str`. Both callers of this method already check that the argument is not none before invoking this method.